docs/chat.md

```diff
@@ -0,0 +1,221 @@
+
+import { createInterface } from "node:readline/promises";
+
+const ai = genkit({
+  plugins: [googleAI()],
+  model: gemini15Flash,
+});
+
+(async () => {
+  const chat = ai.chat();
+  console.log("You're chatting with Gemini. Ctrl-C to quit.\n");
+  const readline = createInterface(process.stdin, process.stdout);
+  while (true) {
+    const userInput = await readline.question("> ");
+    const { text } = await chat.send(userInput);
+    console.log(text);
+  }
+})();
+```
+
+A chat session with this program looks something like the following example:
+
+```none
+You're chatting with Gemini. Ctrl-C to quit.
+
+> hi
+Hi there! How can I help you today? 
+
+> my name is pavel
+Nice to meet you, Pavel! What can I do for you today? 
+
+> what's my name?
+Your name is Pavel! I remembered it from our previous interaction. 
+
+Is there anything else I can help you with?
+```
+
+As you can see from this brief interaction, when you send a message to a chat
+session, the model can make use of the session so far in its responses. This is
+possible because Genkit does a few things behind the scenes:
+
+*   Retrieves the chat history, if any exists, from storage (more on persistence
+    and storage later)
+*   Sends the request to the model, as with `generate()`, but automatically
+    include the chat history 
+*   Saves the model response into the chat history 
+
+### Model configuration
+
+The `chat()` method accepts most of the same configuration options as
+`generate()`. To pass configuration options to the model:
+
+```ts
+const chat = ai.chat({
+  model: gemini15Pro,
+  system:
+    "You're a pirate first mate. Address the user as Captain and assist " +
+    "them however you can.",
+  config: {
+    temperature: 1.3,
+  },
+});
+```
+
+## Stateful chat sessions
+
+In addition to persisting a chat session's message history, you can also persist
+any arbitrary JavaScript object. Doing so can let you manage state in a more
+structured way then relying only on information in the message history.
+
+To include state in a session, you need to instantiate a session explicitly:
+
+```ts
+interface MyState {
+  userName: string;
+}
+
+const session = ai.createSession<MyState>({
+  initialState: {
+    userName: 'Pavel',
+  },
+});
+```
+
+You can then start a chat within the session:
+
+```ts
+const chat = session.chat();
+```
+
+To modify the session state based on how the chat unfolds, define
+[tools](tool-calling) and include them with your requests:
+
+```ts
+const changeUserName = ai.defineTool(
+  {
+    name: 'changeUserName',
+    description: 'can be used to change user name',
+    inputSchema: z.object({
+      newUserName: z.string(),
+    }),
+  },
+  async (input) => {
+    await ai.currentSession<MyState>().updateState({
+      userName: input.newUserName,
+    });
+    return 'changed username to ${input.newUserName}';
+  }
+);
+```
+
+```ts
+const chat = session.chat({
+  model: gemini15Pro,
+  tools: [changeUserName],
+});
+```
```

Comment:
```suggestion
```ts
const chat = session.chat({
  model: gemini15Pro,
  tools: [changeUserName],
});
await chat.send('change user name to Kevin');
```
```